readthedocs/api/v3/views.py

```diff
@@ -126,6 +129,20 @@ class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, ProjectQuerySetMixin,
         'active_versions.last_build.config',
     ]
 
+    def get_serializer_class(self):
+        """
+        Return correct serializer depending on the action.
+
+        For GET it returns a serializer with many fields and on PUT/PATCH/POST,
+        it return a serializer to validate just a few fields.
+        """
+        if self.action in ('list', 'retrieve', 'superproject'):
```

Comment:
what is `superproject` here? They should probably be constants.

Reply:
`superproject` is the `@action` defined in the `ProjectViewSet` that returns the superproject of a project.

I'm not sure if it makes sense to make it a constant. Although, maybe it's good to add a comment in this line to avoid confusions.

Reply:
Yea, we should add a comment at least, since it's not like the others.